addons/web/static/src/core/utils/hooks.js

```diff
@@ -158,12 +158,54 @@ export function useService(serviceName) {
             return result;
         }
     }
-    if (toRaw(service) !== service) {
+    if (!isServiceStateful && autoUseState) {
         return useState(service);
     }
     return service;
 }
 
+/**
+ * Import a service into a component
+ *
+ * @template {keyof import("services").ServiceFactories} K
+ * @param {K} serviceName
+ * @param {Function} [onDelayedServiceReady] - To be called when a delayed stateful service is available
```

Comment:
This is almost always to re-assign the variable, so I would even tell "Used tore-assign the services to variable".